File: src/dashapp/callbacks/viewport.py
"""Viewport clientside callback'i — tarayıcı boyutunu viewport-store'a yazar."""
from __future__ import annotations

import dash
from dash import ClientsideFunction, Input, Output


def register(app: dash.Dash) -> None:
    app.clientside_callback(
        ClientsideFunction(namespace="viewport", function_name="updateViewport"),
        Output("viewport-store", "data"),
        Input("dummy-input", "children"),
    )

File: assets/js/viewport.js
/* Tarayıcı boyutunu viewport-store'a yazar.
 * Dash ClientsideFunction(namespace='viewport', function_name='updateViewport')
 * ile çağrılır.
 */
if (!window.dash_clientside) { window.dash_clientside = {}; }

window.dash_clientside.viewport = {
    updateViewport: function (dummy) {
        return {
            width:  window.innerWidth  || document.documentElement.clientWidth  || document.body.clientWidth,
            height: window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight
        };
    }
};
